rocAL/source/loaders/video/video_loader_sharded.cpp

```diff
@@ -83,6 +83,7 @@ void VideoLoaderSharded::initialize(ReaderConfig reader_cfg, DecoderConfig decod
     // Initialize loader modules
     for (size_t idx = 0; idx < _shard_count; idx++) {
         _loaders[idx]->set_output(_output_tensor);
+        _loaders[idx]->set_gpu_device_id(idx);
```

Comment:
what can the max idx value be? In this function you only check `device_id < 0`. Needs to check upper limit too

Reply:
I think that check needs to happen in higher level when we decide numshards